src/test/scala/holes/IntegrationTests.scala

```diff
@@ -49,22 +59,52 @@ class IntegrationTests extends FunSpec with BeforeAndAfterAll {
   }
 
   describe("produces the expected output") {
-    for (scenario <- Paths.get("src/test/resources").toFile.listFiles().toList.map(_.toPath)) {
-      it(scenario.getFileName.toString) {
-        val expected =
-          new String(Files.readAllBytes(scenario.resolve("expected.txt")), StandardCharsets.UTF_8).trim
-        val actual =
-          compileFile(scenario.resolve("input.scala")).trim
+    for (
+      scenario <- Paths
+        .get("src/test/resources")
+        .toFile
+        .listFiles()
+        .toList
+        .map(_.toPath)
+    ) {
+      expectFile(scalaVersion, scenario) match {
+        case Some(expectedFile) =>
+          it(scenario.getFileName.toString) {
+            val expected =
+              new String(
+                Files.readAllBytes(expectedFile),
+                StandardCharsets.UTF_8
+              ).trim
+            val actual =
+              compileFile(scenario.resolve("input.scala")).trim
 
-        if (actual != expected) {
-          println("Compiler output:")
-          println("=====")
-          println(actual)
-          println("=====")
-        }
-        assert(actual === expected)
+            if (actual != expected) {
+              println("Compiler output:")
+              println("=====")
+              println(actual)
+              println("=====")
+            }
```

Comment:
I'm not convinced about putting specific commands that not everyone, on every system, might have. Otherwise writing the result to a file seems useful.